can-construct.js

```diff
@@ -526,6 +529,49 @@ assign(Construct, {
 	 * dog.speak(); // 'woof'
 	 * blackMamba.speak(); // 'ssssss'
 	 * ```
+	 * 
+	 * ## Alternative value for a new instance
+	 * 
+	 * Sometimes you may want to return some custom value instead of a new object when creating an instance of your class.
+	 * For example, you want your class to act as a singleton, or check whether an item with the given id was already
+	 * created and return an existing one from your cache store (e.g. using [can-connect/constructor/store/store]).
+	 * 
+	 * To achieve this you can return [can-construct.ReturnValue] from `setup` method of your class.
+	 * 
+	 * Lets say you have `myStore` to cache all newly created instances. And if an item already exists you want to merge
+	 * the new data into the existing instance and return the updated instance.
+	 * 
+	 * ```
+	 * const myStore = {};
+	 * 
+	 * const Item = Construct.extend({
+	 *     setup: function(params){
+	 *         if (myStore[params.id]){
+	 *             var item = myStore[params.id];
+	 *             
+	 *             // Merge new data to the existing instance:
+	 *             Object.keys( params ).forEach(function( key ){
+	 *                 item[key] = params[key];
+	 *             });
+	 *             
+	 *             // Return the updated item:
+	 *             return new Construct.ReturnValue( item );
+	 *         } else {
+	 *             return [params];
+	 *         }
+	 *     },
+	 *     init: function(params){
+	 *         this.id = params.id;
+	 *         this.name = params.name;
+	 *         
+	 *         // Save to cache store:
+	 *         myStore[this.id] = this;
```

Comment:
This should almost certainly be done in `setup`.  `init` is usually left for people inheriting from `Item` to use for their particular `Item` type.